src/test/scala/org/ergoplatform/mining/ErgoMinerSpec.scala

```diff
@@ -2,30 +2,41 @@ package org.ergoplatform.mining
 
 import akka.actor.{Actor, ActorRef, ActorSystem, Props}
 import akka.pattern.ask
-import akka.testkit.TestKit
+import akka.testkit.{TestKit, TestProbe}
 import akka.util.Timeout
+import org.bouncycastle.util.BigIntegers
 import org.ergoplatform.local.ErgoMiner.{MiningStatusRequest, MiningStatusResponse, StartMining}
 import org.ergoplatform.local.TransactionGenerator.StartGeneration
-import org.ergoplatform.local.{ErgoMinerRef, TransactionGeneratorRef}
+import org.ergoplatform.local.{ErgoMiner, ErgoMinerRef, TransactionGeneratorRef}
 import org.ergoplatform.mining.Listener._
-import org.ergoplatform.nodeView.state.StateType
+import org.ergoplatform.modifiers.mempool.ErgoTransaction
+import org.ergoplatform.nodeView.ErgoReadersHolder.{GetReaders, Readers}
+import org.ergoplatform.nodeView.history.ErgoHistoryReader
+import org.ergoplatform.nodeView.state.{StateType, UtxoStateReader}
 import org.ergoplatform.nodeView.{ErgoNodeViewRef, ErgoReadersHolderRef}
-import org.ergoplatform.settings.{Algos, ErgoSettings, TestingSettings}
-import org.ergoplatform.utils.ErgoTestHelpers
-import org.scalatest.FlatSpecLike
+import org.ergoplatform.settings.{ErgoSettings, TestingSettings}
+import org.ergoplatform.utils.{ErgoTestHelpers, ValidBlocksGenerators}
+import org.ergoplatform.{ErgoBoxCandidate, Input}
+import org.scalatest.FlatSpec
+import scapi.sigma.DLogProtocol.DLogProverInput
+import scorex.core.ModifierId
+import scorex.core.NodeViewHolder.ReceivableMessages.LocallyGeneratedTransaction
 import scorex.core.network.NodeViewSynchronizer.ReceivableMessages.SemanticallySuccessfulModifier
+import sigmastate.Values.TrueLeaf
+import sigmastate.interpreter.{ContextExtension, ProverResult}
 
 import scala.concurrent.duration._
 import scala.concurrent.{Await, Future}
+import scala.language.postfixOps
 
-class ErgoMinerSpec extends TestKit(ActorSystem()) with FlatSpecLike with ErgoTestHelpers {
+class ErgoMinerSpec extends FlatSpec with ErgoTestHelpers with ValidBlocksGenerators {
 
   val defaultAwaitDuration: FiniteDuration = 5.seconds
   implicit val timeout: Timeout = Timeout(defaultAwaitDuration)
 
   def await[A](f: Future[A]): A = Await.result[A](f, defaultAwaitDuration)
 
-  it should "not freeze while generating candidate block with large amount of txs" in {
+  it should "not freeze while generating candidate block with large amount of txs" in new TestKit(ActorSystem()){
```

Comment:
Why new testkit here?
Also code formatting looks incorrect

Reply:
@catena2w 
Cause we need to guarantee isolation for each case. Sometimes tests being failed with strange errors if we use one system for every case.